add CUDA extension (#2268)
FluxML/Flux.jl: NEWS.md

See also [github's page](https://github.com/FluxML/Flux.jl/releases) for a complete list of PRs merged before each release.

## v0.14.0
* Flux now requires julia v1.9 or later.
* CUDA.jl is not a hard dependency anymore. CUDA support is now provided through the extension mechanism. In order to unlock the CUDA 
functionalities user are required to load CUDA, e.g. with `using CUDA`.

> The package `cuDNN.jl` also needs to be installed in the environment.

Author: @ToucheSir is this phrasing ok?

Reviewer: Yes, just some wordsmithing for your consideration:
```suggestion
* Flux now requires julia v1.9 or later.
* CUDA.jl is no longer a hard dependency. CUDA support is now provided through the [extension](https://pkgdocs.julialang.org/v1/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions)) mechanism.
In order to unlock CUDA functionality, the `CUDA` and `cuDNN` packages must be installed in the environment and CUDA.jl must be loaded, e.g. with `using CUDA`.
```
